meshtastic/interfaces/ble/notifications.py

```
@@ -228,3 +283,422 @@ def _get_callback(self, characteristic: str) -> Callable[[Any, Any], None] | Non
+                        FROMNUM_UUID,
+                        timeout=NOTIFICATION_START_TIMEOUT,
+                    )
+                if attempt + 1 < max_attempts:
+                    _sleep(BLEConfig.SERVICE_CHARACTERISTIC_RETRY_DELAY * (attempt + 1))
+                    continue
+                logger.warning(
+                    "Unable to start FROMNUM notifications for %s after %d attempts due to BlueZ 'Notify acquired'; falling back to polling reads.",
+                    FROMNUM_UUID,
+                    max_attempts,
+                )
+                return
```

Ruff (0.15.6)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)